### PR TITLE
Mk 1710 revise e2e tests for taqueria registered contracts plugin

### PR DESCRIPTION
**Pull request by michaelkernaghan**
# 🌮 Taqueria PR

## 🪁 Description

Revised e2e tests for taqueria registered contracts plugin
did a little cleanup in ligo plugin tests

## 🪂 Pre-Merge Checklist (Definition of Done)

🚦 Required to merge:

- [ ] ⛱️ I have completed this PR template in full and updated the title appropriately
- [ ] ⛵ My code builds cleanly, and I have manually tested the changes
- [ ] 🏄‍♂️ Another team member has built this branch and done manual testing on the change
- [ ] 🏖️ New and existing unit tests pass locally and in CI
- [ ] 🔱 The test plan has been implemented and verified by an SDET
- [ ] 🦀 Automated tests have been written and added to this PR
- [ ] 🐬 I have commented my code, particularly in hard-to-understand areas
- [ ] 🤿 Corresponding changes have been made to all documentation
- [ ] 🐚 Required changes to the VScE have been made
- [ ] 🪸 Required updates to scaffolds have been made
- [ ] 🚢 The release checklist has been completed

## 🛩️ Summary of Changes

Revised e2e tests for taqueria registered contracts plugin

## 🎢 Test Plan
working tets in cicd

## 🛸 Type of Change

- [ ] 🧽 Chore ➾
- [ ] 🛠️ Fix ➾
- [ ] ✨ Feature ➾
- [ x ] 👷 Refactor ➾
- [ ] 🧪 Pre-Release ➾
- [ ] 🚀 Release ➾

### CODE REVIEW

File: tests/e2e/taqueria-registered-contracts.spec.ts [hunk 1]
```diff
@@ -1,183 +1,159 @@
...
+		digest.update(bytes);
+		const hash = digest.digest('hex');
 
-	afterAll(async () => {
-		await rm(taqueriaProjectPath, { force: true, recursive: true });
+		const configFile = await readFile('./test-project/.taq/config.json');
+		const json = JSON.parse(configFile.toString());
+		expect(json).toBeInstanceOf(Object);
+		expect(json).toHaveProperty('contracts');
+		expect(json.contracts).toEqual({
```
Comment (egarson) [inline tests/e2e/taqueria-registered-contracts.spec.ts:27]: What is being tested here?

Comment (egarson) [reply tests/e2e/taqueria-registered-contracts.spec.ts:27]: This seems to be quite tightly tied to the implementation as opposed to testing what the implementation does.
Am I missing something?

File: tests/e2e/taqueria-registered-contracts.spec.ts [hunk 2]
```diff
@@ -1,183 +1,159 @@
...
+
+		await cleanup();
 	});
 
-	test('Initial config has no registered contracts', async () => {
-		const config = await readFile(configFile, { encoding: 'utf-8' });
-		const json = JSON.parse(config);
+	test('Initial config will have no registered contracts', async () => {
+		const { execute, cleanup, readFile, exists } = await prepareEnvironment();
+		const {} = await execute('taq', 'init test-project');
```
Comment (egarson) [inline tests/e2e/taqueria-registered-contracts.spec.ts:72]: elide the empty assigment

File: tests/e2e/taqueria-registered-contracts.spec.ts [hunk 3]
```diff
@@ -1,183 +1,159 @@
...
+		const json = JSON.parse(await readFile('./test-project/.taq/config.json'));
 		expect(json).toBeInstanceOf(Object);
-		expect(json).toHaveProperty('contracts');
-		return expect(json.contracts).toEqual({
-			'hello-tacos.mligo': {
-				sourceFile: 'hello-tacos.mligo',
-				hash,
-			},
-		});
+		expect(json).not.toHaveProperty('contracts');
```
Comment (egarson) [inline tests/e2e/taqueria-registered-contracts.spec.ts:92]: Hmmm this is testing implementation

**Review by egarson — APPROVED** [review]
Lots of implementation testing herein, but I do understand you've inherited this problem.

File: tests/e2e/taqueria-registered-contracts.spec.ts (hunk 2 above)
Comment (michaelkernaghan) [reply tests/e2e/taqueria-registered-contracts.spec.ts:72]: sorry?

Comment (michaelkernaghan) [reply tests/e2e/taqueria-registered-contracts.spec.ts]: ok i see i will elide them all

File: tests/e2e/taqueria-registered-contracts.spec.ts (hunk 3 above)
Comment (michaelkernaghan) [reply tests/e2e/taqueria-registered-contracts.spec.ts:89]: Removed the unnecessary implementation stuff. There are a couple of cases where the config file has to be set up before the test; I had to leave those.

**Pull request merged**